packages/grafana-ui/src/components/Box/Box.tsx

```diff
@@ -0,0 +1,54 @@
+import { css } from '@emotion/css';
+import React, { useCallback } from 'react';
+
+import { GrafanaTheme2, ThemeSpacingTokens } from '@grafana/data';
+
+import { useStyles2 } from '../../themes';
+
+type ExtendSpacing = {
+  vertical?: ThemeSpacingTokens;
+  horizontal?: ThemeSpacingTokens;
+  top?: ThemeSpacingTokens;
+  bottom?: ThemeSpacingTokens;
+  left?: ThemeSpacingTokens;
+  right?: ThemeSpacingTokens;
+};
+type Spacing = ThemeSpacingTokens | ExtendSpacing;
+
+interface BoxProps {
+  margin?: Spacing;
+  padding?: Spacing;
+  backgroundColor?: keyof GrafanaTheme2['colors']['background'];
+}
+
+export const Box = ({ children, ...props }: React.PropsWithChildren<BoxProps>) => {
+  const styles = useStyles2(useCallback((theme) => getStyles(theme, props), [props]));
+
+  return <div className={styles.root}>{children}</div>;
+};
+
+const getSpacing = (theme: GrafanaTheme2, spacing: Spacing = 0) => {
+  if (typeof spacing === 'number') {
+    return theme.spacing(spacing);
+  }
+  const getValue = (prio1: 'top' | 'bottom' | 'left' | 'right', prio2: 'vertical' | 'horizontal') => {
+    return spacing?.[prio1] ?? spacing?.[prio2] ?? 0;
+  };
+
+  return theme.spacing(
+    getValue('top', 'vertical'),
+    getValue('right', 'horizontal'),
+    getValue('bottom', 'vertical'),
+    getValue('left', 'horizontal')
+  );
+};
+
+const getStyles = (theme: GrafanaTheme2, props: BoxProps) => {
+  return {
+    root: css({
+      margin: getSpacing(theme, props.margin),
+      padding: getSpacing(theme, props.padding),
+      backgroundColor: theme.colors.background[props.backgroundColor ?? 'primary'],
+    }),
```

Comment:
i think i would prefer that for these "primative" components where there's more of a direct mapping between react props to css properties, that we use the [array-of-objects form of `css()`](https://github.com/grafana/grafana/blob/f657ab9224f50787eed28bc0070e31cd1a78a376/packages/grafana-ui/src/components/Text/Text.tsx/#L124-L135)

Reply:
this would make it easier to have priorities for the different spacing props:

```ts
  return {
    root: css([
      padding && {
        padding: theme.spacing(padding),
      },
      paddingX && {
        paddingLeft: theme.spacing(paddingX),
        paddingRight: theme.spacing(paddingX),
      },
      paddingY && {
        paddingTop: theme.spacing(paddingY),
        paddingBottom: theme.spacing(paddingY),
      },
      paddingTop && { paddingTop: theme.spacing(paddingTop) },
      paddingBottom && { paddingBottom: theme.spacing(paddingBottom) },
      paddingLeft && { paddingLeft: theme.spacing(paddingLeft) },
      paddingRight && { paddingRight: theme.spacing(paddingRight) },
    ]),
  };
```

Reply:
This would have to be more like 

```
  return {
    root: css([
      typeof padding !== 'undefined && {
        padding: theme.spacing(padding),
      },
      typeof paddingX !== 'undefined && {
        paddingLeft: theme.spacing(paddingX),
        paddingRight: theme.spacing(paddingX),
      },
      ...
  };
```

So that 0 also sets the respective property. Or we could do `padding || padding === 0` or `padding >= 0` to get these to work as well. Do we have any preference for any of these?